src/esthri/src/lib.rs

```diff
@@ -1084,54 +1094,104 @@ where
     Ok(())
 }
 
-async fn sync_local_to_remote<T, P: AsRef<Path>>(
-    s3: &T,
-    bucket: &str,
-    key: &str,
-    directory: P,
-    glob_includes: &[Pattern],
-    glob_excludes: &[Pattern],
-) -> Result<()>
+fn create_dirent_stream<'a>(
+    directory: &'a Path,
+    glob_includes: &'a [Pattern],
+    glob_excludes: &'a [Pattern],
+) -> impl Stream<Item = Result<(String, Option<ListingMetadata>)>> + 'a {
+    async_stream::stream! {
+        for entry in WalkDir::new(directory) {
+            let entry = if let Ok(entry) = entry {
+                entry
+            } else {
+                yield Err(entry.err().unwrap().into());
```

Comment:
Would it make more sense here and below to use a return? I feel like using the break implies other things will happen after breaking from the loop.

Reply:
Changed